app/Http/Controllers/SportsController.php

```diff
@@ -2,31 +2,76 @@
 
 namespace App\Http\Controllers;
 
-use App\Models\Sport;
+use App\Http\Controllers\Controller;
+use App\Http\Requests\StoreCountriesPlaceBySportsRequest;
+use App\Models\CountriesPlaceBySports;
 use App\Models\Country;
-use Illuminate\Http\Request;
+use App\Models\Sport;
+use Illuminate\Support\Facades\DB;
 
 class SportsController extends Controller
 {
+
     public function create()
     {
         $sports = Sport::all();
         $countries = Country::all();
-
         return view('sports.create', compact('sports', 'countries'));
     }
 
-    public function store(Request $request)
+    public function store(StoreCountriesPlaceBySportsRequest $request)
     {
-        // Add your code here
 
+        $parsed_request = $this->parseStoreRequest($request);
+        $parsed_request = $this->addCountryIdToParsedRequest($parsed_request);
+
+        CountriesPlaceBySports::upsert($parsed_request, ["place", "sport_id"], ["country_id"]);
         return redirect()->route('show');
     }
 
     public function show()
     {
-        // Add your code here
 
-        return view('sports.show');
+        $places_by_countries = Country::select(
+            "name",
+            DB::raw("(SELECT count(countries_place_by_sports.id)  from countries_place_by_sports where place = 1 and country_id = countries.id) as gold_count"),
```

Comment:
@hayaIshida this should work but probably could be done shorter with `withCount()`, look at other pull requests and my upcoming video in a few days.